src/App.tsx

```diff
@@ -15,37 +17,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [reverse, setReverse] = useState(false);
+
+  const handleSort = (type: SortType) => {
+    if (sortType === type) {
+      setReverse(curr => !curr);
+    } else {
+      setSortType(type);
+      setReverse(false);
+    }
+  };
+
+  const sortedGoods = [...goodsFromServer].sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => handleSort(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => handleSort(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setReverse(curr => !curr)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || reverse) && (
+          <button
+            type="button"
+            className={cn('button is-danger', {
+              'is-light': !reverse,
+            })}
+            onClick={() => {
+              setSortType(SortType.NONE);
+              setReverse(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here